test/src/tests/generated/body-complex/models.go

```diff
@@ -195,69 +201,79 @@ func (c *Cookiecuttershark) UnmarshalJSON(body []byte) error {
 	return nil
 }
 
-// Datetimerfc1123Wrapper is
+// Datetimerfc1123Wrapper
 type Datetimerfc1123Wrapper struct {
 	autorest.Response `json:"-"`
 	Field             *date.TimeRFC1123 `json:"field,omitempty"`
 	Now               *date.TimeRFC1123 `json:"now,omitempty"`
 }
 
-// DatetimeWrapper is
+// DatetimeWrapper
 type DatetimeWrapper struct {
 	autorest.Response `json:"-"`
 	Field             *date.Time `json:"field,omitempty"`
 	Now               *date.Time `json:"now,omitempty"`
 }
 
-// DateWrapper is
+// DateWrapper
 type DateWrapper struct {
 	autorest.Response `json:"-"`
 	Field             *date.Date `json:"field,omitempty"`
 	Leap              *date.Date `json:"leap,omitempty"`
 }
 
-// DictionaryWrapper is
+// DictionaryWrapper
 type DictionaryWrapper struct {
 	autorest.Response `json:"-"`
 	DefaultProgram    *map[string]*string `json:"defaultProgram,omitempty"`
 }
 
-// Dog is
+// Dog
 type Dog struct {
 	ID   *int32  `json:"id,omitempty"`
 	Name *string `json:"name,omitempty"`
 	Food *string `json:"food,omitempty"`
 }
 
-// DoubleWrapper is
+// DoubleWrapper
 type DoubleWrapper struct {
 	autorest.Response                                                               `json:"-"`
 	Field1                                                                          *float64 `json:"field1,omitempty"`
 	Field56ZerosAfterTheDotAndNegativeZeroBeforeDotAndThisIsALongFieldNameOnPurpose *float64 `json:"field_56_zeros_after_the_dot_and_negative_zero_before_dot_and_this_is_a_long_field_name_on_purpose,omitempty"`
 }
 
-// DurationWrapper is
+// DurationWrapper
 type DurationWrapper struct {
 	autorest.Response `json:"-"`
 	Field             *string `json:"field,omitempty"`
 }
 
-// Error is
+// Error
 type Error struct {
 	Status  *int32  `json:"status,omitempty"`
 	Message *string `json:"message,omitempty"`
 }
 
-// Fish is
-type Fish interface {
+// IFish
+type IFish interface {
 	AsSalmon() (*Salmon, bool)
```

Comment:
Or maybe this should also include `AsIShark()` ?

Reply:
Hm synced with @mcardosos  and @jhendrixMSFT  . A couple of options here. Will discuss them further.